yt_dlp/extractor/niconicochannelplus.py

```diff
@@ -0,0 +1,497 @@
+
+
+class NiconicoChannelPlusIE(NiconicoChannelPlusBaseIE):
+    IE_NAME = 'NiconicoChannelPlus'
+    IE_DESC = 'ニコニコチャンネルプラス'
+    _VALID_URL = r'https?://nicochannel\.jp/(?P<channel>[a-z\d\._-]+)/(?:video|live)/(?P<code>sm\w+)'
+    _TESTS = [{
+        # real video url, normal channel name.
+        'url': 'https://nicochannel.jp/kaorin/video/smsDd8EdFLcVZk9yyAhD6H7H',
+        'info_dict': {
+            'id': 'smsDd8EdFLcVZk9yyAhD6H7H',
+            'title': '前田佳織里はニコ生がしたい！',
+            'ext': 'mp4',
+            'channel': '前田佳織里の世界攻略計画',
+            'channel_id': 'kaorin',
+            'channel_url': 'https://nicochannel.jp/kaorin',
+            'live_status': 'not_live',
+            'thumbnail': 'https://nicochannel.jp/public_html/contents/video_pages/74/thumbnail_path',
+            'description': '２０２１年１１月に放送された\n「前田佳織里はニコ生がしたい！」アーカイブになります。',
+            'timestamp': 1641360276,
+            'duration': 4097,
+            'comment_count': int,
+            'view_count': int,
+            'tags': [],
+            'upload_date': '20220105',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        # real video url, normal channel name, age limited, test purpose channel.
+        'url': 'https://nicochannel.jp/testman/video/smDXbcrtyPNxLx9jc4BW69Ve',
+        'info_dict': {
+            'id': 'smDXbcrtyPNxLx9jc4BW69Ve',
+            'title': 'test oshiro',
+            'ext': 'mp4',
+            'channel': '本番チャンネルプラステストマン',
+            'channel_id': 'testman',
+            'channel_url': 'https://nicochannel.jp/testman',
+            'age_limit': 18,
+            'live_status': 'was_live',
+            'timestamp': 1666344616,
+            'duration': 86465,
+            'comment_count': int,
+            'view_count': int,
+            'tags': [],
+            'upload_date': '20221021',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        # real video url, numbers in channel name.
+        'url': 'https://nicochannel.jp/dateno8noba/video/smVGqtKpdmva4Mcrw7rbeQ8Y',
+        'only_matching': True,
+    }, {
+        # real video url, hyphens in channel name.
+        'url': 'https://nicochannel.jp/owstv-plus/video/smUPTNizUxVspEu5YeDtV3VB',
+        'only_matching': True,
+    }, {
+        # real video url, underscores in channel name.
+        'url': 'https://nicochannel.jp/sakaguchi_kugimiya/video/smieBu2u2kDTYCvYZmLvUaUN',
+        'only_matching': True,
+    }, {
+        # real video url, dots in channel name.
+        'url': 'https://nicochannel.jp/kanase.ito/video/smWCdanZc5bJYMPYhpVp6Sn6',
+        'only_matching': True,
+    }, {
+        # fake live url, normal channel name.
+        'url': 'https://nicochannel.jp/example/live/sm3Xample',
+        'info_dict': {
+            'id': 'sm3Xample',
+            'ext': 'mp4',
+        },
+        'params': {
+            'skip_download': False,
+        },
+        'skip': '404 Not Found',
```

Comment:
How can fake_url have "normal channel name"? And what's even the point of this test?